cv32/tests/core/debug_test/single_step.S

```diff
@@ -112,14 +110,51 @@ _step_trig_exit:
     li t0, 2
     sw t0, 0(a1)
 
-// Loop to wait for interrupt to be handled
-// TODO: If stepie=0, add timeout to below loop 
 _irq_wait_loop:
     la a1, glb_expect_irq_entry
     lw t0, 0(a1);
     bne t0, x0, _irq_wait_loop
-    
-    
+
+
+    //-----------------
+    // Stepping with interrupt, stepie=0
+    addi t0, t0,2 // debug code moves dpc to here
+    li t1, 2
+    // If trigger was correct, debug code skips
+    // loading of t0 to 1, and t0 should be of value 2
+    bne t0, t1, _single_step_fail
```

Comment:
Not sure if these lines are needed here, copy/paste or merge error? Regsiter t0 should contain zero before this code executes, so it shouldn't cause any issue, but maybe remove it for clarity.